vmdb/app/helpers/api_helper/renderer.rb

```diff
@@ -326,9 +326,16 @@ def render_resource_attr(resource, attr)
 
     def physical_attribute_selection(resource)
       return [] unless params['attributes']
-      physical_attributes = params['attributes'].split(",").collect do |attr|
-        attr unless attr_virtual?(resource, attr)
-      end.compact
+
+      @pas ||= {}
+      klass_name = resource.class.name
+      return @pas[klass_name] unless @pas[klass_name].nil?
+
+      @pas[klass_name] = gen_physical_attribute_selection(resource)
+    end
+
+    def gen_physical_attribute_selection(resource)
+      physical_attributes = params['attributes'].split(",").select { |attr| !attr_virtual?(resource, attr) }
```

Comment:
Took a couple of iterations, cause I broke some logic as not all are objects, had to deal with hashes (i.e. tags, etc.).

Updated PR utilizing columns_hash for objects,  re-pushed.

Alberto Bellotti
abellott@redhat.com





> On Feb 25, 2015, at 5:24 PM, Jason Frey <notifications@github.com> wrote:
> 
> In vmdb/app/helpers/api_helper/renderer.rb:
> 
> > @@ -326,9 +326,16 @@ def render_resource_attr(resource, attr)
> >  
> >      def physical_attribute_selection(resource)
> >        return [] unless params['attributes']
> > -      physical_attributes = params['attributes'].split(",").collect do |attr|
> > -        attr unless attr_virtual?(resource, attr)
> > -      end.compact
> > +
> > +      @pas ||= {}
> > +      klass_name = resource.class.name
> > +      return @pas[klass_name] unless @pas[klass_name].nil?
> > +
> > +      @pas[klass_name] = gen_physical_attribute_selection(resource)
> > +    end
> > +
> > +    def gen_physical_attribute_selection(resource)
> > +      physical_attributes = params['attributes'].split(",").select { |attr| !attr_virtual?(resource, attr) }
> 
> what you are calling physical attrbutes are really just columns, right?
> 
> If that's the case, you don't need the cache...can you just lookup with Model.columns_hash (which is cached already)?
> 
> —
> Reply to this email directly or view it on GitHub.
>